### ISSUE TITLE
When reconnecting, network is reachable, despite enabling Persistent tun

### ISSUE THREAD

**Issue by GoogleCodeExporter** [schwabe/ics-openvpn#195]
```
What steps will reproduce the problem?
1. Enable WiFi and connect to VPN;
2. Disable WiFi (to force the VPN to reconnect) and, after mobile network is 
connected and while openVPN is reconnecting, open a site;
3. You have non-vpn internet access

What is the expected output? What do you see instead?
It is supposed to prevent non-vpn network access when reconnecting and 
"persistent tun" is enabled

What mobile phone are you using?
Samsung Galaxy Nexus

Which Android Version and stock ROM or aftermarket like cyanogenmod?
Android 4.3 Google stock ROM

Please provide any additional information below
If you go to a site that shows your IP address with the VPN connected, it shows 
the VPN IP address; if you force the VPN to reconnect (ex. disabling Wifi), and 
refresh the IP address site, it shows your local non-vpn IP; once reconnected, 
it shows the openVPN IP address.
```

Original issue reported on code.google.com by `marcogsi...@gmail.com` on 22 Aug 2013 at 10:20

**Comment by privacy-wolf** [comment]
Just to confirm that I think I experience the same issue (data leaking outside the vpn for a short time when a new connection becomes available). I'm running Android 6. Since this is a potential security breach I would consider this a critical bug (or I might be misunderstanding something).

**Comment by sebkraemer** [comment]
I'm pretty sure I experienced the same a few days ago, where after leaving flight mode I saw the new email notification before the "Connecting.." notification display went through to being connected. Not sure if wifi/3g off-on would be any different under the hood. I think I also saw network connections with another than my later vpnned IP in "Net Monitor" when starting it before coming back from flight mode. but that was another time.

persistent-tun is on, but as far as I understand, it must be recreated once the remote IP changes? Can there be other reasons? I'm trying now to include openvpn-ics in the list of the power-unoptimized apps but no idea if that makes any difference, since the issue is not easy to track.

Is there no way to have a _real_ "kill switch" in an app as it is provided using the native VPN android settings? In my experience, nothing came through when it didn't show as connected but had serious other drawbacks so that I really don't want to use that any more.
